### PR TITLE
Fingerprint Vault

### PR DESCRIPTION
**Pull request by dadgar**

### CODE REVIEW

File: client/fingerprint/vault.go [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+	"github.com/hashicorp/nomad/nomad/structs"
+	vapi "github.com/hashicorp/vault/api"
+)
+
+const (
+	vaultAvailable   = "available"
+	vaultUnavailable = "unavailable"
+)
+
+// ConsulFingerprint is used to fingerprint the architecture
```
Comment (diptanu) [inline client/fingerprint/vault.go]: incorrect comment

File: client/fingerprint/vault.go [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+	client    *vapi.Client
+	lastState string
+}
+
+// NewVaultFingerprint is used to create a Vault fingerprint
+func NewVaultFingerprint(logger *log.Logger) Fingerprint {
+	return &VaultFingerprint{logger: logger, lastState: vaultUnavailable}
+}
+
+func (f *VaultFingerprint) Fingerprint(config *client.Config, node *structs.Node) (bool, error) {
```
Comment (diptanu) [inline client/fingerprint/vault.go]: Comment here

File: client/fingerprint/vault.go [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+		// Print a message indicating that Vault is not available anymore
+		if f.lastState == vaultAvailable {
+			f.logger.Printf("[INFO] fingerprint.consul: Vault is unavailable")
+		}
+		f.lastState = vaultUnavailable
+		return false, nil
+	}
+
+	node.Attributes["vault.accessible"] = strconv.FormatBool(true)
```
Comment (diptanu) [inline client/fingerprint/vault.go]: If we are expecting a min version of the Vault API we should ensure that we do the check during fingerprinting.

Comment (dadgar) [inline client/fingerprint/vault.go]: Will be done server side

**Pull request merged**